pkg/cloud-provider/securitygroup_impl.go

```diff
@@ -89,19 +89,24 @@ func (sg *SecurityGroupImpl) UpdateSecurityGroupMembers(addressGroupIdentifier *
 }
 
 func (sg *SecurityGroupImpl) UpdateSecurityGroupRules(addressGroupIdentifier *securitygroup.CloudResource,
-	ingressRules []*securitygroup.IngressRule, egressRules []*securitygroup.EgressRule) <-chan error {
+	addRules, rmRules, targetRules []*securitygroup.CloudRule) <-chan error {
 	ch := make(chan error)
 
 	go func() {
 		defer close(ch)
 
+		if len(addRules) == 0 && len(rmRules) == 0 {
```

Comment:
avoid calling updateRules if there is no change in the rules. Add a TODO: we can resolve it later